### ISSUE TITLE
make_int_sequence can not make empty list

### ISSUE THREAD

**Issue by jonathanpoelen** [kvasir-io/mpl#31]
```cpp
call<cfe<eager::make_int_sequence>, int_<2> >{} = list<int_<1>,int_<2>>{}
call<cfe<eager::make_int_sequence>, int_<1> >{} = list<int_<1>>{}
call<cfe<eager::make_int_sequence>, int_<0> >{} = list<int_<0>>{} // instead of list<>{}
```

**Comment by chieltbest** [comment]
It seems like this is a very old algorithm which should be replaced with the standard library counterpart, I'll make it a continuation version.

**Comment by chieltbest** [comment]
Fixed in e941eb2cb0a54a0bfb96082ca4611ea2de27fad6
